ci-operator/step-registry/optional-operators/ci/run-bundle/optional-operators-ci-run-bundle-commands.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "Deploying an operator in the bundle format using operator-sdk run bundle command"
+
+# `run bundle` supports three install modes: AllNamespaces, OwnNamespace and SingleNamespace.
+if [[ "${OO_TARGET_NAMESPACES}" != "AllNamespaces" && "${OO_TARGET_NAMESPACES}" != "OwnNamespace" && "${OO_TARGET_NAMESPACES}" != "SingleNamespace" ]]; then
+    echo "install mode should be one of the supported install modes {AllNamespaces, OwnNamespace, SingleNamespace}"
+    exit 1
+fi
+
+TMPDIR=/tmp
+cd $TMPDIR
+
+# handle creation of install namespace if it does not exist
+if [[ "$OO_INSTALL_NAMESPACE" == "create" ]]; then
+    echo "OO_INSTALL_NAMESPACE is 'create': creating new namespace"
+    oc create namespace "$OO_INSTALL_NAMESPACE"
+elif ! oc get namespace "$OO_INSTALL_NAMESPACE"; then
+    echo "OO_INSTALL_NAMESPACE is '$OO_INSTALL_NAMESPACE' which does not exist: creating"
+    oc create namespace "$OO_INSTALL_NAMESPACE"
+else
+    echo "OO_INSTALL_NAMESPACE is '$OO_INSTALL_NAMESPACE'"
+fi
+
+# create the operator bundle using `run bundle` command.
+operator-sdk run bundle "${OO_BUNDLE_IMG}" \
+                        --index-image "${OO_INDEX_IMG}" \
+                        --namespace "${OO_INSTALL_NAMESPACE}" \
+                        --install-mode "${OO_TARGET_NAMESPACES}"
+
+# create artifacts directory to collect all the manifests in yaml format.
+ARTIFACT_DIR=$TMPDIR/artifacts
+mkdir $ARTIFACT_DIR && cd $ARTIFACT_DIR
```

Comment:
All these will only run when `operator-sdk run buncle` succeeds (b/c the `set -o errexit` at L4), is that okay? I guess this collection will be mostly useful when the `run bundle` fails.

Reply:
Yes, everything from here down should be wrapped in a function that gets called in a trap at the top of the script
```sh
set -o nounset
set -o errexit
set -o errtrace # Needed  for function to inherit the trap.
set -o pipefail

function log_resources() {

	POD_ART="$ARTIFACT_DIR/registrypod.yaml"
    echo "Logging Registry pod as $POD_ART"

	...
}

trap 'log_resources' ERR

echo "Deploying an operator in the bundle format using operator-sdk run bundle command"

...
```